### PR TITLE
feat: Update loading at a CommitID to use the StreamLoader

### PR DESCRIPTION
**Pull request by stbrody**
Builds on https://github.com/ceramicnetwork/js-ceramic/pull/2963

### CODE REVIEW

File: packages/core/src/state-management/repository.ts [hunk 1]
```diff
@@ -208,12 +208,20 @@ export class Repository {
     // the timestamp information that will reveal to us that the CACAO didn't actually expire.
     const optsSkippingCACAOChecks = { ...opts, skipCacaoExpirationChecks: true }
     const base$ = await this.load(commitId.baseID, optsSkippingCACAOChecks)
-    const stateAtCommit = await this.stateManager.atCommit(base$, commitId)
```
Comment (stbrody) [inline packages/core/src/state-management/repository.ts:211]: `stateManager.atCommit` will be deleted in the next PR

File: packages/core/src/stream-loading/stream-loader.ts [hunk 2]
```diff
@@ -149,14 +149,15 @@ export class StreamLoader {
       opts
     )
 
-    // If the commitId is now the tip, we're done.
-    if (baseState.log[baseState.log.length - 1].cid.equals(commitId.commit)) {
```
[the remark this answers is not in the document]
Comment (PaulLeCam) [reply packages/core/src/stream-loading/stream-loader.ts:153]: Seems to me we should keep this optimization and update the test accordingly, no?
Are there any specific expectations about the anchor status when loading a stream at a specific commit?

**Review by PaulLeCam — APPROVED** [review]

File: packages/core/src/stream-loading/state-manipulator.ts [hunk 3]
```diff
@@ -36,6 +36,10 @@ export interface ApplyLogToStateOpts {
   throwOnConflict: boolean
 }
 
+export interface ResetStateToCommitOpts {
```
Comment (ukstv) [inline packages/core/src/stream-loading/state-manipulator.ts:39]: Nitpick, as a way to share an idea. Kind of suggestion to think about. Not a critique in any way, as it is half-baked. Feel free to disregard.

TS provides `type` and `interface` constructs. I find it easier to reason about them, if we use `type` for things like types of parameters or shapes of plain JS objects. It is a standalone entity. Inheritance is out of the equation for `type`. If something is declared as an `interface` though, it conveys a meaning that it has to be implemented by some class. That is, if you refactor an interface, beware of the dragons downstream. Would be curious to hear your thoughts about that heuristic someday.

File: packages/core/src/stream-loading/stream-loader.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ukstv) [reply packages/core/src/stream-loading/stream-loader.ts:153]: This feels like an issue to be fixed later. I see that refactoring exercise as a chance to unbundle (now barely manageable) monoliths we have. That's quite a big endeavour. As @PaulLeCam I am not aware about specific expectations. Thus, I am tempted to say, let's disregard that for now, and get back to it later (+properly documenting semantics of atCommit wrt anchor status) when bigger pieces are in place.

**Review by ukstv — APPROVED** [review]

File: packages/core/src/stream-loading/state-manipulator.ts (hunk 3 above)
Comment (stbrody) [reply packages/core/src/stream-loading/state-manipulator.ts:39]: Yeah I've never felt clear on when to use `interface` vs `type`.  That is helpful context for thinking about it.  I'll do a separate follow-up PR to switch all of these over to `type`

File: packages/core/src/stream-loading/stream-loader.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (stbrody) [reply packages/core/src/stream-loading/stream-loader.ts:153]: okay so to be clear @ukstv are you voting for putting the optimization back in and updating the test, or leaving it out and making a note to add the optimization and figure out the semantics later?

**Pull request merged**